### DISCUSSION TITLE
Change highlight start and end on a character-level, change highlight type

### DISCUSSION THREAD

**Discussion by sethhammett** [koreader/koreader#7799] [Ideas]
Whenever I want to move the highlight by characters instead of words or change the type of highlight— lighten, underline, invert— I have to manually edit the metadata file. I had an idea— there could be two rows of buttons when editing the highlight, the first row would move it by word, the second row move it by a character. There would also be a button "Change Type" that would toggle through the highlight types.

Here is a mock-up I made in Photoshop:
![mockup](https://user-images.githubusercontent.com/83931313/120718152-12806c00-c486-11eb-976b-a18998d675ee.png)

(Should this have gone into an issue or are discussions fine for feature requests?)

**Comment by mergen3107** [comment]
You can (de)highlight character by character if you long press the respective button.

As for changing the highlight type for specific highlights, it sounds really great! I recently proposed to have bookmarks categories of any sort (see #7756), and this so far looks the easiest way (lighten, underline, invert). Though I am not sure how difficult it would be to implement
